node.h

```diff
@@ -0,0 +1,65 @@
+#ifndef UAST_NODE_H_
```

Comment:
This is the public library API. It should provide a plain C interface, not a C++ one. That means:

* No C++ header files
* No classes